agent/Cargo.toml

```diff
@@ -35,19 +34,24 @@ serde_json = "1.0.45"
 serde_yaml = "0.8.11"
 serde_derive = "1.0.104"
 akri-shared = { path = "../shared" }
-sxd-document = "0.3.0"
-sxd-xpath = "0.4.0"
 tempfile = "3.1.0"
 tokio = { version = "0.2", features = ["full"] }
 tokio-core = "0.1"
 tonic = "0.1"
 tower = "0.3" 
-udev = "0.4"
+udev = { version = "0.4", optional = true }
 url = "2.1.0"
 uuid = { version = "0.8.1", features = ["v4"] }
-xml-rs = "0.8.0"
-yaserde = "0.3.13"
-yaserde_derive = "0.3.13"
+xml-rs = { version = "0.8.0", optional = true }
+yaserde = { version = "0.3.13", optional = true }
+yaserde_derive = { version = "0.3.13", optional = true }
 
 [build-dependencies]
 tonic-build = "0.1.1"
+
+[features]
+default = ["onvif", "opcua", "udevfeat"]
+
+onvif = ["xml-rs", "yaserde", "yaserde_derive"]
+opcua = ["opcua-client"]
+udevfeat = ["pest", "pest_derive", "udev"]
```

Comment:
i haven't used features before, would it make sense to be consistent (even if it is not needed):

default = ["onvif-feature", "opcua-feature", "udev-feature"]
onvif-feature = ...
opcua-feature = ...
udev-feature = ...

Reply:
Sure, I changed to `-feat` suffix. Usually the feature names doesn't mention the word "feature", e.g. [smoltcp/Cargo.toml](https://github.com/smoltcp-rs/smoltcp/blob/master/Cargo.toml), but consistency >> convention 😄.